mautic/mautic
# Add toggle to links in CKE5, to add external target

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [x ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

[As mentioned in this ticket](https://github.com/mautic/mautic/issues/13362), there is a regression of functionality to do with Links in the text editor. While CKE4 had too many options (many unused), and we can't simply readd ALL the options (CKE5 has an entirely different structure for adding/changing stuff) it seems logical to at least be able to set `target`.
This aims to fix that by simply setting a decorator. 



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. For a new Mautic install (or when using Gitpod), [make sure GrapesJS is installed](https://github.com/mautic/mautic/issues/13510) (recent merges of 5.x seems to have broken this)
3. If you already have a Mautic installed to test this PR in, I'd advise to run `npm install` followed by `npm run build` in the root of the project. Might be that ckeditor needs this.
4. Edit an Page or Email
5. Add a Text block, with some text
6. Select a bit of text and use the buttons of the text editor (CKE) to make it a link
7. You should now have a toggle option to make the target external (this prints `target="_blank"` in the html)

![cke-link-target](https://github.com/mautic/mautic/assets/758957/2895c503-1662-4376-b487-18bac98bc3bb)
